administration/src/components/generation/EakGeneration.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import {EakForm} from "./EakForm";
+import {Button, ButtonGroup} from "@blueprintjs/core";
+import {CardCreationModel} from "../../models/CardCreationModel";
+import {CardType} from "../../models/CardType";
+import "./EakGeneration.css";
+import {AddEakForm} from "./AddEakForm";
+import {INTENT_DANGER} from "@blueprintjs/core/lib/esm/common/classes";
+
+interface Props {
+
+}
+
+interface State {
+    cardCreationModels: CardCreationModel[];
+}
+
+export class EakGeneration extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            cardCreationModels: [EakGeneration.createEmptyCard()]
+        };
+
+        this.addForm = this.addForm.bind(this);
```

Comment:
The preferred way of writing components nowadays is using function components and react hooks.
That also avoids using constructors and binding funcitons to a context.